### ISSUE TITLE
Please add option to disable middle-click paste

### ISSUE THREAD

**Issue by Mr-Pyro** [linuxmint/cinnamon#8450]
```
 * Cinnamon version (cinnamon --version) 
     4.0.9
 * Distribution - (Mint 19.1)
 * Graphics hardware *and* driver used
     Dell G5 15 GTX 1060m ...
 * 64 bit
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
 ```

**Issue**

**Steps to reproduce**
1. Click middle mouse button

**Expected behaviour**
nothing except in special cases like open in new tab when in a web browser 

**Other information**
This "feature" is extremely annoying especially when using software like gravit designer that uses middle mouse click for panning. This causes random snippets of text to be randomly pasted all over the place.

**Comment by icarter09** [comment]
Many thanks for submitting your feature request for Cinnamon. Your feature request has been documented and will be reviewed by the Linux Mint team.

For more information on our workflow and feature requests, read https://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/faq.html.

**Comment by awsms** [comment]
Has it been implemented ?

**Comment by Telofy** [comment]
If it has been implemented, how do I switch off the constant accidental pasting?

**Comment by mikkovedru** [comment]
This is a really irritating feature that greatly hinders or makes it impossible to use other programs. Here is an example with Obsidian:
https://forum.obsidian.md/t/linux-middle-click-paste-is-problematic-and-needs-to-be-automatically-disabled/34717

**Comment by Secret-chest** [comment]
Please add this. For example, in Chromium I use middle-click for closing tabs, but it pastes into my address bar.

**Comment by xenopeek** [comment]
> This is a really irritating feature that greatly hinders or makes it impossible to use other programs.

Middle-click paste is a Xorg feature AFAIK. I don't know of a way to disable it system wide. Some programs support disabling it, like in Firefox you can set middlemouse.paste to false and in LibreOffice you can change the action for the middle mouse button through Tools>Options>LibreOffice>View.

> For example, in Chromium I use middle-click for closing tabs, but it pastes into my address bar.

I can't reproduce that on Linux Mint 20.3 Cinnamon. Middle-click on a tab closes it. It's only if I drift the mouse cursor below the tab label, so over the address bar, that middle-click pastes in the address bar.

**Comment by mixmastamyk** [comment]
Huh?

Middle mouse button paste works great in the terminal and in text fields.

Middle mouse button also works perfectly in Firefox closing tabs and pasting into the URL bar.

Also works great at panning in Gimp and Blender.  Just tried all these programs.

Maybe you've found a broken program, it's possible I suppose.  But this has been working correctly on my machine as long as I can remember.  Since Netscape on Irix in the 90s long.

Maybe file an issue with the broken program?  Looks like that's what they did with Obsidian and it was fixed.

**Comment by mikkovedru** [comment]
@mixmastamyk 
You are missing the point.

Even if it is functionally working as intended, it still doesn't work for me and countless other people, who:
1. don't use this feature
2. don't want this feature, ever
3. would prefer to use the button for something else customizable. How can I make the middle-click bold or translate the word in LibreOffice? I can't, because useless paste functionality is in my way.

But in reality, this feature does not work properly (neither technically nor because of the lack of understanding of what I want to do) and brings all kinds of problems.

**Comment by Telofy** [comment]
I wonder how many people lost their jobs because they were reviewing a public Google doc and pasted company passwords or embarrassing links all over it without noticing when they were just trying to scroll. 

For me the result has been that I've spammed people with notifications of suggestions that I've made to their documents – all nonsense that I pasted by accident often without noticing it. Luckily so far nothing embarrassing or security-relevant.

**Comment by mixmastamyk** [comment]
If you are cutting and pasting passwords, or even able to select them, you're doing it wrong.  They should never be visible in plain text, anywhere.

If you're doing that, the onus is on you and your organization.  It could also be done mistakenly with Ctrl+C and Ctrl+V as well.  I also click the wrong context-menu item once a year or so.  No one seems to be asking to disable those, that I know of.

Still, I have no qualms to have as an option because mistakes happen.  But am still surprised one would give up such a large productivity boost, once learned.  As I mentioned, it does work perfectly in all the programs I've tried recently and in the past.  Maybe something I used was broken in the 90s... can't remember.
